code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java

```diff
@@ -702,6 +706,12 @@ protected Condition inHostList(List<String> hostUUIDs) {
         return HOST.UUID.in(hostUUIDs);
     }
 
+    protected Condition elligibleHostCondition() {
+        return AGENT.ID.isNull().or(AGENT.STATE.eq(CommonStatesConstants.ACTIVE))
```

Comment:
should we add Agent.Removed.isnull to the condition?

Reply:
This method is an extraction from another method so that I could reuse it. Since that logic is long established and apparently not causing any issues, I'd rather not change it.